source/euler/hyperbolic_system.h

```diff
@@ -1313,5 +1339,159 @@ namespace ryujin
+       }
+
+
+       if((pressure.has_ghost_elements() != true) || (density.has_ghost_elements() != true)
+           ||(momentum.at(0).has_ghost_elements() != true) || (momentum.at(1).has_ghost_elements() != true))
+       {
+         //TODO: make this some type of exception
+         std::cout << "ghost elements not allowed " << std::endl;
+         std::cout << "pressure " << (pressure.has_ghost_elements() != true) << std::endl;
+         std::cout << "momx " << (momentum.at(0).has_ghost_elements() != true) << std::endl;
+         std::cout << "momy " << (momentum.at(1).has_ghost_elements() != true) << std::endl;
+         std::cout << "density " << (density.has_ghost_elements() != true) << std::endl;
+
+         exit(EXIT_FAILURE);//TODO: remove, for now, I want this to just quit the program
+       }
+       density.update_ghost_values();
+       pressure.update_ghost_values();
+       for(auto mom: momentum)
+         mom.update_ghost_values();
+
+
+       for(const auto& cell : offline_data_.dof_handler().active_cell_iterators())
+       {
+         if(cell->is_locally_owned())
+         {
+           for(unsigned int face = 0; face < cell->n_faces(); ++face)
+           {
+             if(cell->face(face)->at_boundary()
+                 && cell->face(face)->boundary_id() == ryujin::Boundary::slip)
+             {
+               //if on circle, we do the calculation
+               //first, find if the face center is on the circle
+               const auto tria_iterator = cell->face(face);//->center(true);//center of the face on the manifold
+               const dealii::Point<dim> face_center = tria_iterator->center();
+               const double distance = face_center.distance(disk_center);
+
+   //            std::cout << "face center " << face_center << std::endl;
+   //            std::cout << "distance " << distance << std::endl;
+               //check if we are on circle
+               if(distance < disk_radius + 1e-6)//TODO: ask W for a better way to decide if a face is on circle
+               {
+   //              std::cout << "face center on circle: " << face_center << std::endl;
+                 fe_face_values.reinit(cell, face);
+
+                 //pressure values
+                 fe_face_values.get_function_values(pressure, pressure_values);
+
+                 //now, loop over quadrature points calculating their contribution to the forces acting on the face
+                 for(int q = 0; q < n_q_points; ++q)
+                 {
+                   normal_vector = -fe_face_values.normal_vector(q);
+
+                   //form the contributions from pressure
+                   for(unsigned int d = 0; d < dim; ++d)
+                     fluid_pressure[d][d] = pressure_values[q];
+
+                   fluid_stress = - fluid_pressure;//for the euler equations, the only contribution to stresses comes from pressure
+                   forces = fluid_stress*normal_vector*fe_face_values.JxW(q);
```

Comment:
These two variables only live in this one place. Move the *declaration* here so as to limit the range of code where they are alive.

Reply:
TODO: move declaration much closer to where they are used.